tooling/src/hypothesistooling/projects/hypothesispython.py

```diff
@@ -171,6 +179,17 @@ def build_distribution():
 
 def upload_distribution():
     tools.assert_can_release()
+
+    # This block ensures that Hypothesis 5.0 will drop Python 2 support as close
+    # to midnight as reasonably possible, without being early.  For details see:
+    # https://docs.travis-ci.com/user/customizing-the-build/#build-timeouts
+    if parse_release_file()[0] == "major":
+        at = datetime(2020, 1, 1)
+        while at - timedelta(minutes=40) < datetime.utcnow() < at:
```

Comment:
Suggestion:

1. Move this logic into the general release script. Make it apply for *all* releases near midnight, don't worry about whether it's a major release or not.
2. Use the following code:

```python
release_gap = timedelta(minutes=40)
while True:
    now = datetime.utcnow()
    if now.year == (now + release_gap).year:
        break

    print(f"Waiting for the midnight release...")
    time.sleep(5)    
```

This works regardless of the year and is a bit simpler (which is good for untested code...)